src/processors/default/createERC721NFTsFromTransfersProcessor.ts

```diff
@@ -31,43 +32,68 @@ const processorFunction = (contracts: NftFactory[]) =>
         throw 'buildNFTId not specified'
       }
 
+      const contractAddress = formatAddress(contract.id);
+      const fromAddress = formatAddress(item.args!.from);
+      const toAddress = formatAddress(item.args!.to);
       const tokenId = BigInt((item.args!.tokenId as BigNumber).toString());
-      const newMint = item.args!.from === ETHEREUM_NULL_ADDRESS;
+      const nftId = contractType.buildNFTId(contractAddress, tokenId);
+
       transfers.push({
-        id: `${CHAIN}/${item.blockNumber}/${item.logIndex}`,
-        contractAddress: formatAddress(contract.id),
-        from: item.args!.from,
-        to: item.args!.to,
+        id: ethereumTransferId(item.blockNumber, item.logIndex),
+        contractAddress: contractAddress,
+        from: fromAddress,
+        to: toAddress,
         tokenId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        nftId: contractType.buildNFTId(contract.id, tokenId), 
+        nftId: nftId,
         transactionHash: item.transactionHash
       });
-      if (!newMint) {
-        updates.push({
-          id: contractType.buildNFTId(contract.id, tokenId),
-          owner: item.args!.to
+
+      if (!newMint(fromAddress)) {
+        updatedNFTs.push({
+          id: nftId,
+          owner: toAddress
         })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: fromAddress, amount: -1 })
+        updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
         return undefined;
       }
       newNFTs.push({
-        id: contractType.buildNFTId(contract.id, tokenId),
+        id: nftId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        contractAddress: formatAddress(contract.id),
+        contractAddress: contractAddress,
         tokenId,
         platformId: contract.platformId,
-        owner: item.args!.to,
+        owner: toAddress,
         approved: contract.autoApprove
       });
+      updatedNftsCollectors.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
     });
 
+    // update Collectors table
+    const allCollectors = updatedNftsCollectors.map<Collector>(({ collectorId }) => { return { id: collectorId } } );
+    await clients.db.insert(Table.collectors, _.uniqBy(allCollectors, 'id'), { ignoreConflict: 'id' })
+
+    // fetch any existing NftCollectors
+    const idPairs = _.uniq(updatedNftsCollectors.map( (e) => [e.nftId, e.collectorId] ))
+    const existingNftsCollectors = await clients.db.getRecords<NFTsCollectors>(Table.nftsCollectors,
+      [ ['whereIn', [ ['nftId', 'collectorId'], idPairs ] ] ]
+    );
+
+    // update all NFTsCollectors with most up to date amounts
+    const fullNftsCollectors = updatedNftsCollectors.concat(existingNftsCollectors)
+    const consolidatedNftsCollectors = consolidate(fullNftsCollectors)
+    await clients.db.insert(Table.nftsCollectors, consolidatedNftsCollectors, { ignoreConflict: ['nftId', 'collectorId'] });
```

Comment:
this all looks good. just a small suggestion to change variable naming for ease of readibility/understanding:
 - updatedNftsCollectors is really a set of changes, rather than a set of records, so even though in this case the data type for those changes has the same fields and structure as the data type for the underlying record, they are a little conceptually different
 - i'd create a seperate type to make this explicit, like, `type NFTsCollectorChanges = NFTsCollectors` 
 - instead `updatedNftsCollectors: NFTsCollectors[]` i'd name it something like `nftCollectorChanges: NFTsCollectorChanges[]` and fullNftsCollectorChanges
 - and then just use the name updatedNftsCollectors instead of consolidatedNftsCollectors

Reply:
Yeaaahhh! Much nicer!!!! 💯